framework/include/base/XFEMInterface.h

```diff
@@ -0,0 +1,112 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#ifndef XFEMINTERFACE_H
+#define XFEMINTERFACE_H
+
+#include "ConsoleStreamInterface.h"
+#include "MooseTypes.h"
+#include "MooseVariableBase.h"
+
+class MooseApp;
+class AuxiliarySystem;
+class NonlinearSystem;
+class MaterialData;
+
+namespace libMesh
+{
+  class MeshBase;
+  class QBase;
+}
+
+/**
+ * This is the XFEMInterface class.  This is an abstract base
+ * class that defines interfaces with a class that dynamically
+ * modifies the mesh in support of a phantom node approach for XFEM
+ */
+
+
+// ------------------------------------------------------------
+// XFEMInterface class definition
+class XFEMInterface : public ConsoleStreamInterface
+{
+public:
+
+  /**
+   * Constructor
+   */
+  explicit
+  XFEMInterface(MooseApp & app):
+    ConsoleStreamInterface(app),
+    _material_data(NULL),
+    _mesh(NULL),
+    _mesh2(NULL)
+  {
+  }
+
+  /**
+   * Destructor
+   */
+  virtual ~XFEMInterface()
+  {
+  }
+
+  /**
+   * Set the pointer to the primary mesh that is modified by XFEM
+   */
+  void setMesh(MeshBase* mesh)
+  {
+    _mesh=mesh;
```

Comment:
There are probably a ton of spacing issues to sort out.  Now I know what the policy is, I'll go through and clean all those up.